File: Seminar5/HomeWork/PresenterStart.java
package HomeWork;

import java.util.Scanner;
import java.util.logging.*;

import HomeWork.Controller.PresenterRation;
import HomeWork.View.Menu;

public class PresenterStart {

    PresenterRation presenterRation;
    Menu menu;
    Scanner scanner;
    PresenterStartRation startRation;
    PresenterStartComplex startComplex;

    public PresenterStart(PresenterRation presenterRation,
                          Menu menu,
                          Scanner scanner,
                          PresenterStartRation startRation,
                          PresenterStartComplex startComplex) {
        this.presenterRation = presenterRation;
        this.menu = menu;
        this.scanner = scanner;
        this.startRation = startRation;
        this.startComplex = startComplex;
    }

    public void start() {
        Logger logger = Logger.getGlobal();
        switch (menu.selectCalc()) {
            case 1:
                logger.info("Выбраны рациональные числа");
                startRation.startRation();
                break;
            case 2:
                logger.info("Выбраны комплексные числа");
                startComplex.startComplex();
                break;
            default:
                logger.info("Ошибка");
                System.out.println("Неверный ввод");
                break;

        }
    }

}

File: FinalTaskCalculator/PresenterStart.cs
using System;
using static Controller.PresenterRation;
using static View.Menu;

public class PresenterStart {

    Controller.PresenterRation presenterRation  = null!;
    View.Menu menu = null!;
    PresenterStartRation startRation  = null!;
    PresenterStartComplex startComplex  = null!;

    public PresenterStart(Controller.PresenterRation presenterRation,
                          View.Menu menu,
                          PresenterStartRation startRation,
                          PresenterStartComplex startComplex) {
        this.presenterRation = presenterRation;
        this.menu = menu;
        this.startRation = startRation;
        this.startComplex = startComplex;
    }

    public void Start() {
        switch (menu.SelectCalc()) {
            case 1:
                Console.WriteLine("Выбраны рациональные числа");
                startRation.StartRation();
                break;
            case 2:
                Console.WriteLine("Выбраны комплексные числа");
                startComplex.StartComplex();
                break;
            default:
                Console.WriteLine("Ошибка");
                Console.WriteLine("Неверный ввод");
                break;

        }
    }

}